From 0721addc4dafb1c1a6be5bd1b05b9ebaa90acdc8 Mon Sep 17 00:00:00 2001
From: Jari Aalto <jari.aalto@cante.net>
Date: Fri, 14 Sep 2007 09:31:42 +0300
Subject: [PATCH] Makefile: add install target

Signed-off-by: Jari Aalto <jari.aalto@cante.net>
---
 Makefile |   29 ++++++++++++++++++++++++++---
 1 files changed, 26 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index e71d0ad..ea4ee3e 100644
--- a/Makefile
+++ b/Makefile
@@ -22,8 +22,19 @@ DISTFILES=CHECKSUM.S FILES FLIP.MAN FLIP.PRJ INSTALL.DOC MAKEFILE.TCC	\
 
 # Where to install executable and manual files on "make install".  The trailing
 # "/." forces an error message if the destination directory doesn't exist.
-BINDIR = /usr/local/bin/.
-MANDIR = /usr/man/man1/.
+
+DESTDIR         =
+prefix          = /usr
+exec_prefix     = $(prefix)
+man_prefix      = $(prefix)/share
+
+INSTALL         = /usr/bin/install
+INSTALL_BIN     = $(INSTALL) -m 755
+INSTALL_DATA    = $(INSTALL) -m 644
+INSTALL_OBJS    = flip
+
+MANDIR          = $(DESTDIR)$(man_prefix)/man/man1
+BINDIR          = $(DESTDIR)$(exec_prefix)/bin
 
 # CC is compiler, LD is loader (may be same as compiler), CFLAGS are flags
 # for compiler, LDFLAGS are flags for loader, CFMORE are additional
@@ -43,6 +54,8 @@ OBJS = flip.o
 #getopt.o: getopt.c flip.h
 #	$(CC) $(CFLAGS) $(CFMORE) $*.c
 
+all: sys_v
+
 nothing:
 	@echo \
 	'Please type "make sys_v", "make bsd", "make uport", or "make ultrix"'
@@ -68,7 +81,17 @@ flip.o: flip.c flip.h
 clean:
 	rm -f *.o core flip
 
-install:
+install-bin:
+	$(INSTALL_BIN) -d $(BINDIR)
+	$(INSTALL_BIN) -s $(INSTALL_OBJS) $(BINDIR)
+
+install-man:
+	$(INSTALL_BIN) -d $(MANDIR)
+	$(INSTALL_DATA) *.1 $(MANDIR)
+
+install: all install-bin install-man
+
+install-old:
 	mv flip $(BINDIR)
 	cp flip.1 $(MANDIR)
 
-- 
1.5.3

